### ISSUE TITLE
buffer tensor serialization when  using  paddle.jit.save

### ISSUE THREAD

**Issue by OccupyMars2030** [PaddlePaddle/Paddle#45460]
### 请提出你的问题 Please ask your question

Please refer to my AI studio project:
https://aistudio.baidu.com/aistudio/projectdetail/4465624

**Comment by weisy11** [comment]
![image](https://user-images.githubusercontent.com/19523330/186878129-88db511a-f5fc-44fc-8e52-361d493d0625.png)
Please ask your question

**Comment by OccupyMars2030** [comment]
<img width="646" alt="image" src="https://user-images.githubusercontent.com/31559413/186884026-acb96f06-87be-4bef-97cf-b84ed5037518.png">

The question is located at  the last cell.

**Comment by OccupyMars2030** [comment]
Just for  my reference: 
The code example is used  at  two  locations  but  with  different  emphasis:
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/jit/case_analysis_cn.html#to-tensor
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/jit/principle_cn.html#buffer

**Comment by weisy11** [comment]
不好意思，没有明白你想问什么。
你是想指出文档中有错误，还是说使用过程中遇到了bug？如果你遇到了bug，请提供复现的代码，谢谢。
btw，如果你的项目没有发布的话，我是看不到你的修改的。

**Comment by OccupyMars2030** [comment]
抱歉，我不太会用aistudio，现在我的项目公开了，应该看得到了
<img width="272" alt="d77e5c5e4ef674139a58fcd1de39312" src="https://user-images.githubusercontent.com/31559413/187192723-050271ff-3f7c-4303-adb1-ef024ea78ee0.png">

不是bug， 只是单纯地学习文档时，有一部分内容没有理解，想请飞桨专家为我讲解一下，谢谢

**Comment by weisy11** [comment]
错误的用法可能在一些情况下也可能得到正确的结果。但是并不意味着在其他情况下也能正确运算。
文档的意思是，self.mask 不能正常进行参数的更新。例如在你的例子中，如果这样修改：
```
class SimpleNet(paddle.nn.Layer):
    def __init__(self, mask):
        super(SimpleNet, self).__init__()
        self.linear = paddle.nn.Linear(10, 3)
        self.mask = mask

    @to_static(input_spec=[InputSpec([None, 10], name='x'), InputSpec([None, 3], name='y')])
    def forward(self, x, y):
        out = self.linear(x)
        out = out + y
        mask = paddle.to_tensor(self.mask)
        self.mask = [1, 0, 0]   # 在这里更新了mask，你会发现mask并没有正确的更新
        out = out * mask
        return out
```
但如果将mask改成paddle.to_tensor就可以正常更新了。
在使用动转静时，会将forward过程中用到的参数也进行序列化，但是这些参数相当于是一个快照，不能正常进行更新。

**Comment by weisy11** [comment]
错误的用法可能在一些情况下也可能得到正确的结果。但是并不意味着在其他情况下也能正确运算。
文档的意思是，self.mask 不能正常进行参数的更新。例如在你的例子中，如果这样修改：
```
class SimpleNet(paddle.nn.Layer):
    def __init__(self, mask):
        super(SimpleNet, self).__init__()
        self.linear = paddle.nn.Linear(10, 3)
        self.mask = mask

    @to_static(input_spec=[InputSpec([None, 10], name='x'), InputSpec([None, 3], name='y')])
    def forward(self, x, y):
        out = self.linear(x)
        out = out + y
        mask = paddle.to_tensor(self.mask)
        self.mask = [1, 0, 0]   # 在这里更新了mask，你会发现mask并没有正确的更新
        out = out * mask
        return out
```
但如果将mask改成paddle.to_tensor就可以正常更新了。
在使用动转静时，会将forward过程中用到的参数也进行序列化，但是这些参数相当于是一个快照，不能正常进行更新。
所以文档里说，正确的用法是把参数都写成tensor
